code/modules/antagonists/malf_ai/malf_ai_modules.dm

```diff
@@ -1014,6 +1024,223 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
+		return FALSE
+
+	var/obj/machinery/vending/clicked_vendor = clicked_on
+
+	if (clicked_vendor.tilted)
+		clicked_vendor.balloon_alert(ai_caller, "already tilted!")
+		return FALSE
+
+	if (!clicked_vendor.tiltable)
+		clicked_vendor.balloon_alert(ai_caller, "cannot be tilted!")
+		return FALSE
+
+	if (!clicked_vendor.powered() || clicked_vendor.machine_stat & BROKEN)
+		clicked_vendor.balloon_alert(ai_caller, "no power!")
+		return FALSE
+
+	var/picked_dir_string = show_radial_menu(ai_caller, clicked_vendor, GLOB.all_radial_directions, custom_check = CALLBACK(src, PROC_REF(radial_check), caller, clicked_vendor))
+	if (isnull(picked_dir_string))
+		return FALSE
+	var/picked_dir = get_dir_from_string(picked_dir_string)
+
+	var/turf/target = get_step(clicked_vendor, picked_dir)
+	if (!ai_caller.can_see(target))
+		to_chat(ai_caller, span_warning("You can't see the target tile!"))
+		return FALSE
+
+	new /obj/effect/temp_visual/telegraphing/vending_machine_tilt(target, time_to_tilt)
+	clicked_vendor.visible_message(span_warning("[clicked_vendor] starts falling over..."))
+	clicked_vendor.balloon_alert_to_viewers("falling over...")
+	addtimer(CALLBACK(src, PROC_REF(do_vendor_tilt), clicked_vendor, target), time_to_tilt)
+
+	adjust_uses(-1)
+	if(uses)
+		desc = "[initial(desc)] It has [uses] use\s remaining."
+		build_all_button_icons()
+
+	unset_ranged_ability(caller, span_danger("Tilting..."))
+	return TRUE
+
+/datum/action/innate/ai/ranged/remote_vendor_tilt/proc/do_vendor_tilt(obj/machinery/vending/vendor, turf/target)
+	if (QDELETED(vendor))
+		return FALSE
+
+	if (vendor.tilted || !vendor.tiltable)
+		return FALSE
+
+	var/crit = (rand(1, 100) >= (100 - MALF_VENDOR_TIPPING_CRIT_CHANCE)) //mimicks vendor chances
+
+	vendor.tilt(target, crit)
+
+/// Used in our radial menu, state-checking proc after the radial menu sleeps
+/datum/action/innate/ai/ranged/remote_vendor_tilt/proc/radial_check(mob/living/silicon/ai/caller, obj/machinery/vending/clicked_vendor)
+	if (caller.incapacitated() || caller.stat == DEAD || QDELETED(caller))
+		return FALSE
+
+	if (QDELETED(clicked_vendor))
+		return FALSE
+
+	if (uses <= 0)
+		return FALSE
+
+	if (!caller.can_see(clicked_vendor))
+		to_chat(caller, span_warning("Lost sight of [clicked_vendor]!"))
+		return FALSE
+
+	return TRUE
+
+/datum/ai_module/utility/core_tilt
+	name = "Rolling Servos"
+	description = "Allows you to slowly roll around, crushing anything in your way with your bulk."
+	cost = 20
+	one_purchase = FALSE
+	power_type = /datum/action/innate/ai/ranged/core_tilt
+	unlock_sound = 'sound/effects/bang.ogg'
+	unlock_text = "You gain the ability to roll over and crush anything in your way."
+
+/datum/action/innate/ai/ranged/core_tilt
+	name = "Roll over"
+	button_icon_state = "roll_over"
+	desc = "Allows you to roll over in the direction of your choosing, crushing anything in your way."
+	auto_use_uses = FALSE
+	ranged_mousepointer = 'icons/effects/mouse_pointers/supplypod_target.dmi'
+	uses = 20
```

Comment:
remove the uses mechanic and just make this a purchase upgrade that lasts forever

Reply:
no, ais shouldnt get unrestricted movement around the station

Reply:
now as a delta-tier goal? yes. but as a upgrade they can get at any time? no

Reply:
if the low uses given is a problem i can increase em but its not a permanent upgrade

Reply:
(increase uses or decrease cost pick your poison)

Reply:
it's slow as balls and literally telegraphs the movement, it's fine, remove the uses feature and make it a permanent upgrade

Reply:
> it's slow as balls and literally telegraphs the movement, it's fine, remove the uses feature and make it a permanent upgrade

it isnt the actual lethality of the thing that is the power, it is the fact the core can just move around unrestricted
im really not interested in giving ais the on-demand ability to just negate one of the biggest things _about_ ai: the inability to move much

i can increase the uses given or decrease the cost (or a combination of both) but i really think this isn't somehting that should be a permanent upgrade

if a maintainer disagrees with me or agrees with you i'd be more willing to change it but as of now no

Reply:
can't they just have a borg or shell move themselves anyway?

moving this way is slow, makes really loud CLANGs, and immediately tells anyone who sees your core that you're a malf AI